dask/array/routines.py

```diff
@@ -263,6 +271,26 @@ def rot90(m, k=1, axes=(0, 1)):
         return flip(transpose(m, axes_list), axes[1])
 
 
+def _tensordot_rechunk(array, inner_axes, limit=None):
+    """Optimize rechunking for tensordot computation."""
+    if limit is None:
+        limit = parse_bytes(config.get("array.chunk-size"))
```

Comment:
I don't know if I should make this limit smaller or not. Sometimes I do see yellow memory bars on the workers.

Right now I fit the chunk size for each array to this limit, but I don't take into account if we need memory to do the processing, or even how the chunk shapes of one array will interact with the chunk shapes of the second array. (I'm not strictly opposed to "magic numbers", but I'd be guessing wildly)

Reply:
Here's an example, I can make the worker memory bars go yellow more often by increasing the size of the outer dimensions:

```python
import dask.array as da
x = da.random.random((1000, 1000, 500), chunks=(50, 50, 50))
y = da.random.random((1000, 1000), chunks=(50, 50))
axes = (1, 0)
z = da.tensordot(x, y, axes=axes, rechunk=True)
```

I suspect maybe I'll need a combination of the two approaches I've tried (i.e. check both the size of the chunks in the input arrays, and check for the size of the chunks in the output array)